File: languages/js/src/client/bloock.ts
import { ConfigData } from "../bridge/proto/bloock_config";
import { NewConfigData } from "../config/config";
import { AuthenticityClient } from "./authenticity";
import { AvailabilityClient } from "./availability";
import { EncryptionClient } from "./encryption";
import { IdentityClient } from "./identity";
import { IdentityCoreClient } from "./identity_core";
import { IntegrityClient } from "./integrity";
import { KeyClient } from "./key";
import { RecordClient } from "./record";
import { WebhookClient } from "./webhook";

/**
 * Represents a client for interacting with the Bloock SDK.
 */
export class BloockClient {
  private configData: ConfigData;

  public AuthenticityClient: AuthenticityClient;
  public AvailabilityClient: AvailabilityClient;
  public EncryptionClient: EncryptionClient;
  public IdentityCoreClient: IdentityCoreClient;
  public IdentityClient: IdentityClient;
  public IntegrityClient: IntegrityClient;
  public KeyClient: KeyClient;
  public RecordClient: RecordClient;
  public WebhookClient: WebhookClient;

  /**
   * Creates a new instance of the Bloock SDK client with the specified configuration.
   * @param apiKey
   * @param identityApiHost
   */
  constructor(apiKey: string, identityApiHost?: string) {
    this.configData = NewConfigData(undefined);

    if (this.configData.config) {
      this.configData.config.apiKey = apiKey;
      this.configData.config.identityApiHost = identityApiHost;
    }

    this.AuthenticityClient = new AuthenticityClient(this.configData);
    this.AvailabilityClient = new AvailabilityClient(this.configData);
    this.EncryptionClient = new EncryptionClient(this.configData);
    this.IdentityCoreClient = new IdentityCoreClient(this.configData);
    this.IdentityClient = new IdentityClient(this.configData);
    this.IntegrityClient = new IntegrityClient(this.configData);
    this.KeyClient = new KeyClient(this.configData);
    this.RecordClient = new RecordClient(this.configData);
    this.WebhookClient = new WebhookClient(this.configData);
  }
}

File: languages/go/client/bloock.go
// Package client provides a client for interacting with the Bloock SDK.
package client

import (
	"github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"
	"github.com/bloock/bloock-sdk-go/v2/internal/config"
)

// BloockClient represents a client for interacting with the Bloock SDK.
type BloockClient struct {
	configData *proto.ConfigData

	AuthenticityClient AuthenticityClient
	AvailabilityClient AvailabilityClient
	EncryptionClient   EncryptionClient
	IdentityCoreClient IdentityCoreClient
	IdentityClient     IdentityClient
	IntegrityClient    IntegrityClient
	KeyClient          KeyClient
	RecordClient       RecordClient
	WebhookClient      WebhookClient
}

// NewBloockClient creates a new instance of the Bloock SDK client with the specified configuration.
func NewBloockClient(apiKey string, identityApiHost *string, forceEnv *string) BloockClient {
	configData := config.NewConfigDataDefault()
	configData.Config.ApiKey = apiKey

	if forceEnv != nil {
		configData.Config.Environment = forceEnv
	}

	if identityApiHost != nil {
		configData.Config.IdentityApiHost = identityApiHost
	}

	return BloockClient{
		configData: configData,

		AuthenticityClient: NewAuthenticityClientWithConfig(configData),
		AvailabilityClient: NewAvailabilityClientWithConfig(configData),
		EncryptionClient:   NewEncryptionClientWithConfig(configData),
		IdentityCoreClient: NewIdentityCoreClientWithConfig(configData),
		IdentityClient:     NewIdentityClientWithConfig(configData),
		IntegrityClient:    NewIntegrityClientWithConfig(configData),
		KeyClient:          NewKeyClientWithConfig(configData),
		RecordClient:       NewRecordClientWithConfig(configData),
		WebhookClient:      NewWebhookClientWithConfig(configData),
	}
}
